test/Microsoft.TemplateEngine.Cli.UnitTests/AddReferencePostActionTests.cs

```diff
@@ -148,5 +149,50 @@ public void AddRefFindsMultipleDefaultProjFilesInAncestorOfOutputDirectory()
             IReadOnlyList<string> projFilesFound = actionProcessor.FindProjFileAtOrAbovePath(_engineEnvironmentSettings.Host.FileSystem, outputBasePath, new HashSet<string>());
             Assert.Equal(2, projFilesFound.Count);
         }
+
+        [Fact(DisplayName = nameof(AddRefCanHandleProjectFileRenames))]
+        public void AddRefCanHandleProjectFileRenames()
+        {
+            AddReferencePostActionProcessor actionProcessor = new AddReferencePostActionProcessor();
+
+            string targetBasePath = _engineEnvironmentSettings.GetNewVirtualizedPath();
+            string projFileFullPath = Path.Combine(targetBasePath, "MyApp.csproj");
+            string referencedProjFileFullPath = Path.Combine(targetBasePath, "NewName.csproj");
+            
+            var args = new Dictionary<string, string>() { { "targetFiles", "[\"MyApp.csproj\"]" }, { "referenceType", "project" }, { "reference", "./OldName.csproj" } };
+            var postAction = new MockPostAction { ActionId = AddReferencePostActionProcessor.ActionProcessorId, Args = args };
+
+            var creationEffects = new MockCreationEffects()
+                .WithFileChange(new MockFileChange("./OldName.csproj", "./NewName.csproj", ChangeKind.Change))
```

Comment:
```suggestion
                .WithFileChange(new MockFileChange("./OldName.csproj", "./NewName.csproj", ChangeKind.Create))
```

nit: iirc `ChangeKind.Change` is not used.

Reply:
changed the value to `ChangeKind.Create`. I also added an additional unit test for the case where the referenced project file was not renamed (the case that also worked previously)